release_tooling/src/github_metadata.py

```diff
@@ -0,0 +1,35 @@
+import requests
+
+
+class GitHubMetadata:
+    def __init__(self, github_repository, api_url='https://api.github.com'):
+        self.api_url = api_url
+        self.github_repository = github_repository
+
+    def find_pull_requests(self, commit_ref):
+        repo = ""
+        if self.github_repository:
+            repo = self.github_repository
+
+        params = {
+            'q': f"SHA={commit_ref}&repo:{repo}"
+        }
+
+        response = requests.get(
```

Comment:
Minor: you might want to call `raise_for_status()` on this response, so you error out if this call fails.

(We seem to be at the limit of unauth'd GitHub API requests in the building.)